### DISCUSSION TITLE
How to import binary file

### DISCUSSION THREAD

**Discussion by PierreR** [cue-lang/cue#1279] [Q&A]
I am trying to use `cue import binary` that has been introduced here:

https://cue-review.googlesource.com/c/cue/+/9570

If I try to match the example found here:
https://cue-review.googlesource.com/c/cue/+/9570/2/cmd/cue/cmd/testdata/script/import_binary.txt

I can get something working with `cue import binary --ext crt .` 

Two questions. The first one is why a tabulation before the input and before the second `'''`:
```
-- out/expect.cue --
'''
⇥1234
⇥'''
```
The second question is more related to what I really want to do which is to import a single file; the equivalent of `cue import -f -p ocp  -l 'elk: crt:' --outfile test2.cue test.txt` but for a binary base64 encoded file.

I am trying  `cue import  binary --ext base64 -f -p ocp  -l 'elk: jks:' --outfile test2.cue test.bin.base64` but without success.  I am under the impression there is a bug somewhere but before filling in an issue I would love to have some confirmation.

Thanks for your help

**Comment by antong** [comment]
For the first question, the contents for the file out/expect.cue represents a CUE value of type bytes, with the value being all the bytes in the file imported by cue import binary. There are many ways to format such a bytes literal, for instance it could be `'1234'`. But the binary import uses the "multiline byte sequence" format. Quoting the language specification:

> Multiline strings and byte sequences respectively start with a triple double quote (""") or triple single quote ('''), immediately followed by a newline, which is discarded from the string contents. The string is closed by a matching triple quote, which must be by itself on a newline, preceded by optional whitespace. The newline preceding the closing quote is discarded from the string contents. The whitespace before a closing triple quote must appear before any non-empty line after the opening quote and will be removed from each of these lines in the string literal. A closing triple quote may not appear in the string. To include it is suffices to escape one of the quotes.

So the tabulators and first and last newlines are not part of the value, they are just for formatting. If you were to remove them, everything would work the same, but  `cue fmt` would add them back.

For the second question, I think `cue import binary` can not work on individual files named as arguments, only on all files with the file extension given with the --ext option. So, if you remove the "test.bin.base64" from the command line, and do `cue import binary --ext base64 -f -p ocp -l 'elk: jks:' --outfile test2.cue`, then it would work.
